### PR TITLE
Fix error on mkfs for first partition

### PR DESCRIPTION
**Pull request by landerweit-phytec**
Creating the filesystem on the first partition could fail when `mkfs` is called before the partition is available. Fix this by waiting for the partitions to be ready after writing the partition layout.
Also, add the output to stderr to the error message.

### CODE REVIEW

File: src/utils.c [hunk 1]
```diff
@@ -21,24 +21,25 @@ pu_spawn_command_line_sync(const gchar *command_line,
...
+    if (!g_spawn_sync(NULL, argv, NULL, spawn_flags, NULL, NULL, NULL, &errmsg,
                       &wait_status, error)) {
         g_prefix_error(error, "Failed spawning process: ");
         g_strfreev(argv);
         return FALSE;
     }
 
     if (!g_spawn_check_exit_status(wait_status, error)) {
-        g_prefix_error(error, "Command '%s' failed: ", command_line);
+        g_prefix_error(error, "Command '%s' failed with error message: %s ",
```
Comment (mschwan-phytec) [inline src/utils.c:41]: Maybe put some single quotes around the error message string to distinguish it from the rest of the following error output.

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]

File: src/emmc.c [hunk 2]
```diff
@@ -231,6 +234,12 @@ pu_emmc_setup_layout(PuFlash *flash,
 
     ped_disk_commit(self->disk);
 
+    udevadm_cmd = g_strdup_printf("udevadm settle --timeout %d", UDEVADM_SETTLE_TIMEOUT);
+    if (!g_spawn_command_line_sync(udevadm_cmd, NULL, NULL, &wait_status, error))
+        return FALSE;
+    if (!g_spawn_check_exit_status(wait_status, error))
+        return FALSE;
+
```
Comment (mschwan-phytec) [inline src/emmc.c:242]: Maybe it's worth separating this into its own function. Although there is currently only one place where it is used, it makes sense from a logical point of view.

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]

File: src/emmc.c (hunk 2 above)
Comment (landerweit-phytec) [reply src/emmc.c:242]: Sure, I'll move it to utils.

File: src/utils.c [hunk 3]
```diff
@@ -349,6 +352,18 @@ pu_is_drive(const gchar *device)
...
 }
 
+gboolean
+pu_wait_for_partitions(GError **error)
+{
+    g_autofree gchar *udevadm_cmd = NULL;
+    udevadm_cmd = g_strdup_printf("udevadm settle --timeout %d", UDEVADM_SETTLE_TIMEOUT);
+
+    if (!pu_spawn_command_line_sync(udevadm_cmd, error))
+        return FALSE;
```
Comment (mschwan-phytec) [inline src/utils.c:362]: Why did you remove the check for the exit status here?

File: src/utils.c [hunk 4]
```diff
@@ -14,31 +14,34 @@
 #include "error.h"
 #include "utils.h"
 
+#define UDEVADM_SETTLE_TIMEOUT     10
```
Comment (mschwan-phytec) [inline src/utils.c:17]: Reduce spacing of `UDEVADM_SETTLE_TIMEOUT` and `10` to one space.

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]

File: src/utils.c (hunk 3 above)
Comment (landerweit-phytec) [reply src/utils.c:362]: Since this function is now in utils, I switched to using `pu_spawn_command_line_sync()` which already checks for the exit status.

Comment (mschwan-phytec) [reply src/utils.c:362]: Ah, okay. I overlooked that. :slightly_smiling_face:

**Review by mschwan-phytec — APPROVED** [review]
